config/configuration_file.go

```
@@ -0,0 +1,130 @@
+	"github.com/sonatype-nexus-community/go-sona-types/ossindex/types"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type ChequeConfig struct {
+	OSSIndexConfig OSSIConfig
+	IQConfig IQConfig
+}
+
+type OSSIConfig struct {
+	Username string `yaml:Username`
+	Password string `yaml:Password`
+}
+
+type IQConfig struct {
+	Server   string `yaml:"Server"`
+	Username string `yaml:"Username"`
+	Token    string `yaml:"Token"`
+}
+
+type Config struct {
+	logger *logrus.Logger
+}
+
+func New(logger *logrus.Logger) *Config {
+	return &Config{logger: logger}
+}
+
+func (c Config) CreateOrReadConfigFile() ChequeConfig {
+
+	createDirectory(c.logger, types.IQServerDirName)
+	createDirectory(c.logger, types.OssIndexDirName)
+
+	if !fileExists(getIQConfig()) {
+		writeDefaultIQConfig(c.logger)
+	}
+	if !fileExists(getOssiConfig()) {
+		writeDefaultOssiConfig(c.logger)
+	}
+
+	return readConfig(c.logger)
+}
+
+func createDirectory(logger *logrus.Logger, directory string) {
+	home, _ := os.UserHomeDir()
+	fileDirectory := filepath.Join(home, directory)
+	err := os.MkdirAll(fileDirectory, os.ModePerm)
+	if err != nil {
+		logger.Error("Couldn't create config directory: " + fileDirectory)
+	}
+}
+
+//Gets the default location for the config file
+func getIQConfig() string {
+	home, _ := os.UserHomeDir()
+	filePath := filepath.Join(home,types.IQServerDirName, types.IQServerConfigFileName)
+	return filePath
+}
+
+func getOssiConfig() string {
+	home, _ := os.UserHomeDir()
+	filePath := filepath.Join(home,types.OssIndexDirName, types.OssIndexConfigFileName)
+	return filePath
+}
+
+func readConfig(logger *logrus.Logger) ChequeConfig{
+	iqBytes, err := ioutil.ReadFile(getIQConfig())
+	if err != nil {
+		logger.Error(err)
+	}
+	iqConfig := IQConfig{}
+	yaml.Unmarshal(iqBytes, &iqConfig)
+
+	ossiBytes, err := ioutil.ReadFile(getOssiConfig())
+	if err != nil {
+		logger.Error(err)
+	}
+	ossiConfig := OSSIConfig{}
+	yaml.Unmarshal(ossiBytes, &ossiConfig)
```

We tend to ignore errors when we do calls like this, but `yaml.Unmarshall` totally returns an error if one occurs. Right now you are more or less ignoring that error. If you want to do that, but make it more explicit to others that you are a BOSS level coder and you thought about everything, you can do:

```
_ = yaml.Unmarshall(ossiBytes, &ossiConfig)
```

That just sends a signal to someone looking at your code that either you copy and pasted an example (lol) or that you chose by intention to ignore the error